frontend/lib/resolution/Resolver.cpp

```diff
@@ -2212,11 +2260,18 @@ void Resolver::resolveIdentifier(const Identifier* ident,
     return;
   }
 
+  QualifiedType subType;
+  bool potentialSuper = isPotentialSuper(ident, &subType);
+
   // lookupIdentifier reports any errors that are needed
-  auto vec = lookupIdentifier(ident, receiverScopes);
+  auto vec = lookupIdentifier(ident, receiverScopes, potentialSuper);
 
   if (vec.size() == 0) {
-    result.setType(QualifiedType());
+    if (potentialSuper) {
+      result.setType(getSuperType(context, std::move(subType), ident));
+    } else {
+      result.setType(QualifiedType());
+    }
```

Comment:
In the else-branch of this chain we have this check to see if the one-and-only ID is "empty", which indicates some special behavior. What do you think of returning an empty ID from ``lookupIdentifier`` for ``super``? It seems like it could be useful if we needed to call ``lookupIdentifier`` in some other context, where we could at least know that something was found rather than forgetting to check ``isPotentialSuper``.

I do think we'd want to use ``getSuperType`` instead of trying to modify ``typeForBuiltin``.